Paws4Me/Paws4Me/Display And Save Pets/Repository/SavedPetRepository.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  Repository.swift
+//  Paws4Me
+//
+//  Created by Kirsty-Lee Walker on 2022/03/25.
+//
+
+import Foundation
+import UIKit
+
+class SavedPetDataRepository {
+
+    // MARK: - Vars/Lets
+    typealias PetFetchSavedResult = (Result<[Pet], LocalDatabaseError>) -> Void
+    typealias SavePetResults = (Result<[Pet], LocalDatabaseError>) -> Void
+    private let context = (UIApplication.shared.delegate as? AppDelegate)?.persistentContainer.viewContext
+    private var pets: [Pet]? = []
+    private var namePet = ""
+    private var imagePet = ""
+
+    // MARK: - Fetch local database data
+    func fetchSavedPets(completionHandler: @escaping PetFetchSavedResult) {
+        do {
+            self.pets = try context?.fetch(Pet.fetchRequest())
+            guard let savedPets = self.pets else { return }
+            completionHandler(Result.success(savedPets))
+        } catch _ as NSError {
+            completionHandler(Result.failure(.retrievedPetsSavedError))
+        }
+    }
+
+    // MARK: - Save pets in local database
+    func savePets(namePet: String, imagePet: String, completionHandler: SavePetResults ) {
+        guard let petContext = self.context else { return }
+        let petObject = Pet(context: petContext)
+        petObject.petName = namePet
+        petObject.petImage = imagePet
+
+        do {
+            guard let petContext = self.context else { return }
+            guard let savedPets = self.pets else { return }
```

Comment:
You can add something like this here 
guard let foo = bar, qux = taco else {
  ...
}

Reply:
Thank you for the suggestion 👍🏻